include/system/analytics_manager.js

```diff
@@ -181,30 +399,48 @@ module.exports = function AnalyticsManagerModule(pb) {
      * providers.
      */
     AnalyticsManager.onPageRender = function(req, session, ls, cb) {
-        var site = pb.RequestHandler.sites[req.headers.host] ? pb.RequestHandler.sites[req.headers.host].uid : null;
-        if (checkForHook(site)) {
-            //short circuit when there are not tasks for the site.
-            return cb(null, '');
-        }
-        var managerInstance = new AnalyticsManager({ site: site });
+        var context = {
+            site: pb.RequestHandler.sites[req.headers.host] ? pb.RequestHandler.sites[req.headers.host].uid : null,
+            timeout: pb.config.analytics.timeout
+        };
+        var managerInstance = new AnalyticsManager(context);
         managerInstance.gatherData(req, session, ls, cb);
     };
 
-    function checkForHook(site) {
-        return !PROVIDER_HOOKS[pb.SiteService.GLOBAL_SITE] ||
-            Object.keys(PROVIDER_HOOKS[pb.SiteService.GLOBAL_SITE]).length === 0
-        !PROVIDER_HOOKS[site] ||
-        Object.keys(PROVIDER_HOOKS[site]).length === 0;
-    }
```

Comment:
This function was removed because it was redundant to getKeys because it had to iterate over the hash to create the key array.  Instead, I took the approach of attempting to get the keys then decide based on the length since it had to call ```Object.keys``` anyway.